### ISSUE TITLE
`float_cmp` could catch float literals in `match` expressions

### ISSUE THREAD

**Issue by Centri3** [rust-lang/rust-clippy#11555]
### Summary

rust-lang/rust#116098 removes the `illegal_floating_point_literal_pattern` lint. A float literal in a `match` expression is semantically equivalent to `==` and, if this is merged, I think it makes sense to suggest a guard.

If this lint is kept, this should be closed

### Lint Name

`float_cmp`

### Reproducer

I tried this code:

```rust
#![warn(clippy::float_cmp)]

fn main() {
    let x = 1.2331f64;
    let y = 1.2332f64;
    match x {
        1.23f64 => println!("a"),
        _ => {},
    }
}
```

I expected to see this happen:

Clippy suggests:

```rust
#![warn(clippy::float_cmp)]

fn main() {
    let x = 1.2331f64;
    let y = 1.2332f64;
    match x {
        x if (x - 1.23f64).abs() < f64::EPSILON => println!("a"),
        _ => {},
    }
}
```

Instead, this happened: No warning (from clippy)

### Version

```text
rustc 1.74.0-nightly (bdb0fa3ee 2023-09-19)
binary: rustc
commit-hash: bdb0fa3ee5ffb4cc1a1b53cbe8344a2b83b1ae2a
commit-date: 2023-09-19
host: x86_64-unknown-linux-gnu
release: 1.74.0-nightly
LLVM version: 17.0.0
```
